### PR TITLE
Update Klog dependency

### PR DESCRIPTION
**Pull request by rf232**
**What type of PR is this?**

/kind cleanup
/kind feature

**What this PR does / why we need it**:
Updating Klog to pickup changes needed for log sanitization.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/1753-logs-sanitization
```
/milestone v1.20

### CODE REVIEW

**Comment by serathius** [comment]
Looks like klog added new flag `one-output` 
We need to update tests: 
* `staging/src/k8s.io/component-base/cli/globalflag/go_default_test`
* `staging/src/k8s.io/legacy-cloud-providers/azure/go_default_test`

**Comment by dims** [comment]
@rf232 @serathius there's a new flag `one-output` please see the failed CI job.

**Comment by rf232** [comment]
> Looks like klog added new flag `one-output`
> We need to update tests:
> 
> * `staging/src/k8s.io/component-base/cli/globalflag/go_default_test`
> * `staging/src/k8s.io/legacy-cloud-providers/azure/go_default_test`

Updated flags test.  
Azure test looks to be a flake (second run of same test passes)

**Comment by serathius** [comment]
ping @smarterclayton

**Comment by dims** [comment]
/approve
/lgtm
/assign @liggitt 

(for root approval)

File: vendor/k8s.io/klog/v2/klog.go [hunk 1]
```diff
@@ -752,18 +772,24 @@ func (l *loggingT) printWithFileLine(s severity, logr logr.Logger, file string,
...
 
 // if loggr is specified, will call loggr.Error, otherwise output with logging module.
-func (l *loggingT) errorS(err error, loggr logr.Logger, msg string, keysAndValues ...interface{}) {
+func (l *loggingT) errorS(err error, loggr logr.Logger, filter LogFilter, msg string, keysAndValues ...interface{}) {
+	if filter != nil {
+		msg, keysAndValues = filter.FilterS(msg, keysAndValues)
+	}
 	if loggr != nil {
-		loggr.Error(err, msg, keysAndValues)
+		loggr.Error(err, msg, keysAndValues...)
```
Comment (liggitt) [inline vendor/k8s.io/klog/v2/klog.go:780]: is this a bug fix or does this represent a signature/behavior change?

File: vendor/k8s.io/klog/v2/klog.go [hunk 2]
```diff
@@ -752,18 +772,24 @@ func (l *loggingT) printWithFileLine(s severity, logr logr.Logger, file string,
...
 
 // if loggr is specified, will call loggr.Info, otherwise output with logging module.
-func (l *loggingT) infoS(loggr logr.Logger, msg string, keysAndValues ...interface{}) {
+func (l *loggingT) infoS(loggr logr.Logger, filter LogFilter, msg string, keysAndValues ...interface{}) {
+	if filter != nil {
+		msg, keysAndValues = filter.FilterS(msg, keysAndValues)
+	}
 	if loggr != nil {
-		loggr.Info(msg, keysAndValues)
+		loggr.Info(msg, keysAndValues...)
```
Comment (liggitt) [inline vendor/k8s.io/klog/v2/klog.go:792]: same question - bugfix or behavior change?

File: vendor/k8s.io/klog/v2/klog.go [hunk 3]
```diff
@@ -919,18 +945,22 @@ func (l *loggingT) output(s severity, log logr.Logger, buf *buffer, file string,
...
-			case errorLog:
-				l.file[errorLog].Write(data)
-				fallthrough
-			case warningLog:
-				l.file[warningLog].Write(data)
-				fallthrough
-			case infoLog:
-				l.file[infoLog].Write(data)
+			if l.oneOutput {
+				l.file[s].Write(data)
```
Comment (liggitt) [inline vendor/k8s.io/klog/v2/klog.go:949]: is it possible for s to be something other than fatalLog, errorLog, warningLog or infoLog? previously, values of `s` other than those were ignored

File: vendor/k8s.io/klog/v2/klog.go (hunk 1 above)
Comment (serathius) [reply vendor/k8s.io/klog/v2/klog.go:780]: This is a bugfix https://github.com/kubernetes/klog/pull/186

File: vendor/k8s.io/klog/v2/klog.go (hunk 2 above)
Comment (serathius) [reply vendor/k8s.io/klog/v2/klog.go:792]: https://github.com/kubernetes/klog/pull/186

File: vendor/k8s.io/klog/v2/klog.go (hunk 3 above)
Comment (serathius) [reply vendor/k8s.io/klog/v2/klog.go:949]: No, severity is internal alias for int32. Users of klog cannot pass it's value, it's inferred from function call `klog.Info` calls subfunction with severity `infoLog`.

This code changes is meant to change fallthrough logic. Previously we wrote to files with severtity equal lower to one passed. `one-output` is meant to change this behavior and allow writing to only files matching severity. 
https://github.com/kubernetes/klog/pull/141

File: staging/src/k8s.io/component-base/cli/globalflag/globalflags_test.go [hunk 4]
```diff
@@ -57,7 +57,7 @@ func TestAddGlobalFlags(t *testing.T) {
 	}{
 		{
 			// Happy case
-			expectedFlag:  []string{"add-dir-header", "alsologtostderr", "help", "log-backtrace-at", "log-dir", "log-file", "log-file-max-size", "log-flush-frequency", "logtostderr", "skip-headers", "skip-log-headers", "stderrthreshold", "v", "vmodule"},
+			expectedFlag:  []string{"add-dir-header", "alsologtostderr", "help", "log-backtrace-at", "log-dir", "log-file", "log-file-max-size", "log-flush-frequency", "logtostderr", "one-output", "skip-headers", "skip-log-headers", "stderrthreshold", "v", "vmodule"},
```
Comment (liggitt) [inline staging/src/k8s.io/component-base/cli/globalflag/globalflags_test.go:60]: not thrilled about the global flag that is not obviously prefixed or related to klog / log output

**Comment by liggitt** [comment]
/lgtm
/approve

@dims, can you take an issue to the klog repo to ensure any future flags get a klog or log-specific prefix?

**Comment by dims** [comment]
@liggitt will do!

**Pull request merged**